src/rpdk/core/contract/resource_client.py

```diff
@@ -267,9 +267,40 @@ def generate_invalid_create_example(self):
         example = self.invalid_strategy.example()
         return override_properties(example, self._overrides.get("CREATE", {}))
 
+    def get_unique_keys_for_model(self, create_model):
+        return {
+            k: v
+            for k, v in create_model.items()
+            if self.is_property_in_path(k, self.primary_identifier_paths)
+            or self.is_property_in_path(k, self._additional_identifiers_paths)
+        }
+
+    # Disable the no-self-use violations in this function
+    # pylint: disable=no-self-use
+    def is_property_in_path(self, prop, paths):
+        for path in paths:
+            for key in path:
+                if prop == key:
+                    return True
+        return False
+
+    def validate_update_example_keys(self, unique_identifiers, update_example):
```

Comment:
might be a good idea to use it as a `@decorator` - [lib](https://github.com/aws-cloudformation/cloudformation-cli/blob/master/src/rpdk/core/contract/suite/contract_asserts.py)